earth_enterprise/src/fusion/autoingest/AssetGenDaemonCpp.pl

```diff
@@ -420,68 +420,17 @@ ()
     void AddConfig(DOMElement *parent, const $config &config);
 }
 
-std::shared_ptr<${name}AssetImplD>
-${name}AssetImplD::Load(const std::string &boundref)
+std::string ${name}AssetImplD::GetName() const
 {
-    // make sure the base class loader actually instantiated one of me
-    // this should always happen, but there are no compile time guarantees
-    auto loaded = ${name}AssetVersionImpl::Load(boundref);
-    std::shared_ptr<${name}AssetImplD> result =
-        std::dynamic_pointer_cast<${name}AssetImplD>(loaded);
-
-    if (result == nullptr) {
-        std::string error {"Internal error: "};
-        if (loaded == nullptr)
-           error += "base did not load and ";
-        error += "${name}AssetImplD loaded wrong type for ";
-        AssetThrowPolicy::FatalOrThrow(error + boundref);
-    }
-
-    return result;
+    return "${name}Asset";
 }
 
-extern void ToElement(DOMElement *elem, const AssetStorage &self);
-
-bool
-${name}AssetImplD::Save(const std::string &filename) const
+void ${name}AssetImplD::SerializeConfig(DOMElement *top) const
```

Comment:
Minor: just throwing ideas out there. If we had functions that translated between generic XML and something like a key/value store we could update AddConfig so that it gets its data from the key/value store and not directly from the XML objects. That way only the serializer needs to know about XML.

Reply:
I really like that idea, though at this point I'd rather not include it in this PR. I'll put a comment above the SerializeConfig definitions in Asset.h and AssetVersion.h as a note for future development. I think the usefulness of that change would come into play more if/when different methods of serialization are added.

Reply:
Works for me.